src/generator/writers/writeStdlib.ts

```diff
@@ -1133,6 +1133,32 @@ export function writeStdlib(ctx: WriterContext): void {
         });
     });
 
+    ctx.fun(`__tact_log2_floor_unsafe`, () => {
+        ctx.signature(`int __tact_log2_floor_unsafe(int num)`);
+        ctx.context("stdlib");
+        ctx.asm("", "UBITSIZE");
+    });
+
+    ctx.fun(`__tact_sqrt`, () => {
```

Comment:
how about we implement this in Tact?

Reply:
if a function can be implemented efficiently in Tact, we should implement it in Tact or open an issue and fix it

Reply:
done